### PR TITLE
RAG tutorial

### PR DESCRIPTION
**Pull request by jgolebiowski**

### CODE REVIEW

File: rag-tutorial/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+* Python 3.10+
+* `pip install vllm langchain-core langchain_community langchain-openai langchain-huggingface wikipedia pandas requests rich pyyaml`
+* A **distil labs** account for fine‑tuning
+
+## One‑line demo
+
+```bash
+python -m vllm.entrypoints.openai.api_server \
+    --model HuggingFaceTB/SmolLM2-135M-Instruct --device cpu &
```
Comment (uazafar) [inline rag-tutorial/README.md]: Below in the notebook you are using:
```
2>&1 | tee chat-model.log
```
Does this need to be here too?

File: rag-tutorial/rag-tutorial.ipynb [hunk 2]
```diff
@@ -0,0 +1,2454 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ae6dc736-7f5b-4578-8a64-b8f5a6a588ed",
+   "metadata": {},
+   "source": [
+    "# Chapter 2: Specialize an SLM for local RAG\n",
+    "\n",
+    "In this chapter we will fine‑tune a small language model with **distil labs** in three steps:\n",
```
Comment (uazafar) [inline rag-tutorial/rag-tutorial.ipynb:335]: I'm not sure if this information is elsewhere on slides maybe, but it is not clear at this point of the notebook, why we are moving onto finetuning the small model. Is there some performance metric we have which shows poor results for the non fine-tune case?

**Comment by uazafar** [comment]
Leaving some comments here as they are a bit more generic.

- Do we have any reasons for why we would use an SLM in a RAG setup? What is the selling point? As far as I'm aware, a common setup for RAG is to just hook in ChatGPT/Claude and use that as the underlying LLM. RAG apps are generally not high throughput so what would be the advantage of using an SLM?

- On a similar point to above, what would be the performance difference in using a fine-tuned SLM vs hooking up an LLM like ChaTGPT to the RAG workflow. I think this could be a question from an attendee who may have attempted using RAG before. Not sure if we have an answer to this already but definitely something to prepare for at one point.

- Do we have anywhere else, perhaps in your slides some information about the performance of the non fine-tuned vs fine-tuned SLM? It wasn't clear going through the notebook but I'm wondering if this is in your presentation.

File: rag-tutorial/README.md (hunk 1 above)
Comment (jgolebiowski) [reply rag-tutorial/README.md]: Updated the line to make it more consistent

File: rag-tutorial/rag-tutorial.ipynb (hunk 2 above)
Comment (jgolebiowski) [reply rag-tutorial/rag-tutorial.ipynb:335]: Hmmm good point, we dont really have one ATM beyond qualitative evaluation. I will try to add something there.
